Tasks/AzureRmWebAppDeployment/deploymentProvider/PublishProfileWebAppDeploymentProvider.ts

```diff
@@ -0,0 +1,144 @@
+        if(applyFileTransformFlag) {
+            await this.ApplyFileTransformation();
+        }
+
+        var msDeployPublishingProfile = await this.publishProfileUtility.GetTaskParametersFromPublishProfileFile(this.taskParams);
+        var deployCmdFilePath = this.GetDeployCmdFilePath();
+
+        await this.SetMsdeployEnvPath();
+        var cmdArgs:string = this.GetDeployScriptCmdArgs(deployCmdFilePath, msDeployPublishingProfile);
+
+        var retryCountParam = tl.getVariable("appservice.msdeployretrycount");
+        var retryCount = (retryCountParam && !(isNaN(Number(retryCountParam)))) ? Number(retryCountParam): DEFAULT_RETRY_COUNT; 
+        
+        try {
+            while(true) {
+                try {
+                    retryCount -= 1;
+                    await this.publishProfileUtility.RunCmd(cmdArgs);
+                    break;
+                }
+                catch (error) {
+                    if(retryCount == 0) {
+                        throw error;
+                    }
+                    console.log(error);
+                    console.log(tl.loc('RetryToDeploy'));
+                }
+            }
+            console.log(tl.loc('PackageDeploymentSuccess'));
+        }
+        catch (error) {
+            tl.error(tl.loc('PackageDeploymentFailed'));
+            tl.debug(JSON.stringify(error));
+            msDeployUtility.redirectMSDeployErrorToConsole();
+            throw Error(error.message);
+        }
+        finally {
+            this.ResetMsdeployEnvPath();
+            if(applyFileTransformFlag) {
+                this.ResetFileTransformation();
+            }
+        }
+    }
+
+    public async UpdateDeploymentStatus(isDeploymentSuccess: boolean){ }
+
+    private async SetMsdeployEnvPath() {
+        var msDeployPath = await msDeployUtility.getMSDeployFullPath();
+        var msDeployDirectory = msDeployPath.slice(0, msDeployPath.lastIndexOf('\\') + 1);
+        this.origEnvPath = process.env.PATH;
+        process.env.PATH = msDeployDirectory + ";" + process.env.PATH ;
+    }
+
+    private async ResetMsdeployEnvPath() {
+        process.env.PATH = this.origEnvPath;
+    }
+
+    private GetDeployCmdFilePath(): string {
+        var webPackagePath = packageUtility.PackageUtility.getPackagePath(this.taskParams.Package);
+        var packageDir = path.dirname(webPackagePath);
+        return packageUtility.PackageUtility.getPackagePath(packageDir + "\\*.deploy.cmd");
+    }
+
+    private GetDeployScriptCmdArgs(deploycmdFileName:string, msDeployPublishingProfile:any): string {
+        var deployCmdArgs: string = " /C " + deploycmdFileName + " /Y /A:basic \"/U:" + msDeployPublishingProfile.userName + "\" \"\\\"/P:" + msDeployPublishingProfile.userPWD 
+            + "\\\"\" \"\\\"/M:" + "https://" + msDeployPublishingProfile.publishUrl + "/msdeploy.axd?site=" + this.taskParams.WebAppName + "\\\"\"";
+
```

Comment:
You can improve this logic by appending the site query string only if webAppPath is present

Reply:
In the existing msdeploy scenario, the task always append site query. 
Also in the publish profile utility file, I have made these parameters has to be mandatory present in publish profile.
Please let me know the scenarios when app path will not be present.